ai_services/anomaly_detection/sensitivity_auto_tuner/standalone_univariate/sensitivity_auto_tuner.py

```diff
@@ -0,0 +1,147 @@
+        else:
+            # If there is an entry that satisfies FPR <= des_fpr , then we apply the function of TPR * (1 - FPR)
+            # and sort for the function
+            best_fpr['function'] = best_fpr.apply(lambda row: row['TPR'] * (1 - row['FPR']), axis=1)
+            results = best_fpr.sort_values('function', ascending=False)
+            results = results.head(1)
+        print(f"Optimal Sensitivity to use: {results['Sensitivity'].iloc[0]}\n")
+        print(f"TPR:{results['TPR'].iloc[0]} , FPR: {results['FPR'].iloc[0]}")
+    else:
+        print("No values match desired metrics")
+
+
+def setup(des_tpr: float,
+          des_fpr: float,
+          label_df: pd.DataFrame,
+          ad_client: AnomalyDetectionClient,
+          window_size: int):
+
+    # Generate sensitivity samples
+    sensitivity_samples = np.arange(0, 1, 0.01)
+    # sensitivity_samples = [0.6601652074029946, 0.23891188415967424]
+
+    # Prepare inline request data
+    signal_names = 'value'
+    value = []
+    for index, row in label_df.iterrows():
+        requestData = {"timestamp": row['timestamp'], "value": row['value']}
+        value.append(UnivariateInlineSignalValueRequestData(**requestData))
+
+    univariate_inline_signal_request_data = UnivariateInlineSignalRequestData(signal_name=signal_names, values=value)
+    results = pd.DataFrame(columns=['TPR',
+                                    'FPR',
+                                    'Sensitivity'])
+
+    # Perform inference for each sensitivity sample
+    total_start_time = time.time()
+    for sensitivity_sample in sensitivity_samples:
+        res = get_inference_results(label_df=label_df,
+                                    window_size=window_size,
+                                    sensitivity=sensitivity_sample,
+                                    ad_client=ad_client,
+                                    request=univariate_inline_signal_request_data)
+        results = pd.concat([results, pd.DataFrame.from_records([res])], ignore_index=True)
+
+    total_end_time = time.time()
+    print(f"Total time taken: {total_end_time - total_start_time}")
+    tune_sensitivity(results, des_tpr, des_fpr)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description='Sensitivity Auto Tuner')
+    parser.add_argument('--des_tpr', type=float, default=0.80)
+    parser.add_argument('--des_fpr', type=float, default=0.15)
+    parser.add_argument('--dataset_path', type=str, default='~/')
+    parser.add_argument('--window_size', type=int, default=10)
+    parser.add_argument('--config_path', type=str, default='~/.oci/config')
+    args = parser.parse_args()
+
+    # Setting up AnomalyDetectionClient
+    config = oci.config.from_file(args.config_path)
+    ad_client = AnomalyDetectionClient(config)
+
+    # Read input dataset into DataFrame
+    label_df = pd.read_csv(args.dataset_path)
+
+    setup(args.des_tpr, args.des_fpr, label_df, ad_client, args.window_size)
```

Comment:
If user wants to directly call our `setup` function to integrate with their notebook, ideally, it should contain a `float` object as return for sensitivity, something like:
`optimal_sensitivity = setup(args.des_tpr, args.des_fpr, label_df, ad_client, args.window_size)`
Can we add this? Also make sure in the function you declare the return object type. Thanks.

Reply:
Done